src/neo-vm/Types/CompoundType.cs

```diff
@@ -22,14 +22,14 @@ protected CompoundType(ReferenceCounter referenceCounter)
 
         public abstract void Clear();
 
-        public override int GetHashCode()
+        public sealed override bool GetBoolean()
         {
-            throw new NotSupportedException();
+            return true;
         }
 
-        public override bool ToBoolean()
+        public override int GetHashCode()
         {
-            return true;
+            throw new NotSupportedException();
```

Comment:
We should return 0 instead of an exception in order to force `Equal` comparison

Reply:
`Equals` won't call this `GetHashCode()`. Only `HashSet` and `Dictionary` use it.

Reply:
And I didn't modify this place in this PR.